Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.AwsSignedSubjectToken.cs

```diff
@@ -0,0 +1,279 @@
+
+                DateTime utcNow = clock.UtcNow;
+                string host = verificationRequestUrl.Host;
+                // The service name is the first part of the host, for instance
+                // iam in iam.amazonaws.com or sts in sts.amazonaws.com
+                string serviceName = host.Split(new char[] { '.' }, StringSplitOptions.RemoveEmptyEntries)[0];
+                IDictionary<string, string> canonicalHeaders = CreateCanonicalHeaders();
+                var sortedHeaders = canonicalHeaders.OrderBy(pair => pair.Key, StringComparer.Ordinal);
+                var signedHeaders = string.Join(";", sortedHeaders.Select(pair => pair.Key));
+                // See point 3 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-string-to-sign.html
+                string credentialScope = $"{utcNow.ToString(ShortDateFormat)}/{region.Region}/{serviceName}/{AwsRequestType}";
+
+                string hashedCanonicalRequest = GetHashedCanonicalRequest();
+                string stringToSign = GetStringToSign();
+                string signature = GetSignature();
+                string authorizationHeader = GetAuthorizationHeader();
+
+                return new AwsSignedSubjetToken(verificationRequestUrl.ToString(), StVerificationHttpMethod, "",
+                    canonicalHeaders
+                        .Select(pair => new AwsSignedSubjectTokenHeader(pair.Key, pair.Value))
+                        .Concat(Enumerable.Repeat(new AwsSignedSubjectTokenHeader("Authorization", authorizationHeader), 1))
+                        .ToArray());
+
+                string GetHashedCanonicalRequest()
+                {
+                    // Step 1
+                    // https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+
+                    // First, the HTTP Method.
+                    // See point 1 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    StringBuilder canonicalRequest = new StringBuilder(StVerificationHttpMethod).Append(newLine);
+
+                    // Now the URI's absolute path.
+                    // See point 2 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(verificationRequestUrl.GetPathForAwsCanonicalRequest()).Append(newLine);
+
+                    // Now the query string.
+                    // See point 3 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(verificationRequestUrl.GetQueryStringForAwsCanonicalRequest()).Append(newLine);
+
+                    // Now the canonical headers.
+                    // We just built them oursevels, so we know header names are lower case and that values
+                    // that could have leading or trailing spaces have been trimmed.
```

Comment:
What about "Convert sequential spaces in the header value to a single space."? I suspect we'll never see that, but it would be worth either doing or documenting why we're not.